session-replay/Sources/DatadogSessionReplay/Processor/Flattening/NodesFlattener.swift

```diff
@@ -0,0 +1,43 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import CoreGraphics
+
+/// Flattens VTS received from `Recorder` by transforming its tree-structure of nodes into array of nodes.
+///
+/// Flattening includes removal of nodes that are invisible because of being covered by other nodes displayed
+/// closer to the screen surface.
+internal struct NodesFlattener {
+    /// This current implementation is greedy and works in `O(n*log(n))`, wheares `O(n)` is possible.
+    /// TODO: RUMM-2461 Improve flattening performance.
```

Comment:
There are yet no unit tests for `NodesFlattener` as it is still not in its final shape 🚧. The `RUMM-2461` will focus solely on flattening algorithm and then tests will become mandatory.